src/lib/components/artist/header.tsx

```diff
@@ -52,29 +54,35 @@ class Header extends React.Component<HeaderProps, State> {
     })
   }
 
-  handleFollowChange = () => {
+  @eventHandler("Follow Artist")
+  handleFollowChange() { // this now needs to be a normal method, not a property
     const newFollowersCount = this.state.following ? (this.state.followersCount - 1) : (this.state.followersCount + 1)
-    ARTemporaryAPIModule.setFollowArtistStatus(!this.state.following, this.props.artist._id, (error, following) => {
-      if (error) {
-        console.error(error)
-      } else {
-        Events.postEvent(this, {
-          name: following ? "Follow artist" : "Unfollow artist",
-          artist_id: this.props.artist._id,
-          artist_slug: this.props.artist.id,
-          // TODO At some point, this component might be on other screens.
-          source_screen: "artist page",
-        })
-      }
-      this.setState({ following, followersCount: newFollowersCount })
+    return new Promise((resolve, reject) => {
```

Comment:
I just read through the [decorators typescript docs](https://www.typescriptlang.org/docs/handbook/decorators.html). There isn't a way to enforce a return type on method decorator. My initial thought was using the reflection API in order to get the return type of the method at runtime

However, based on the Promise/A+ spec, we don't need to enforce the method to return a `Promise`. just use `Promise.resolve(valueOrPromiseItDoesn'tMatter)`, you can read more about the logic behind that [here](http://stackoverflow.com/a/27746324)

Useful Links:
- http://blog.wolksoftware.com/decorators-reflection-javascript-typescript
- http://stackoverflow.com/a/27746324